### PR TITLE
test(link): add token theming tests

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #7180

## Summary
Add token theming tests for `link`.

### CODE REVIEW

File: packages/calcite-components/src/components/link/link.e2e.ts [hunk 1]
```diff
@@ -320,4 +322,47 @@ describe("calcite-link", () => {
...
+            shadowSelector: "span",
+            targetProp: "color",
+          },
+        ],
+        "--calcite-link-background-color": [
+          {
+            shadowSelector: "span",
+            targetProp: "backgroundColor",
+          },
+        ],
```
Comment (alisonailea) [inline packages/calcite-components/src/components/link/link.e2e.ts:340]: nitpick. `--calcite-link-text-color` & `--calcite-link-background-color` don't have to be arrays

**Review by alisonailea — APPROVED** [review]
Nice

**Review by jcfranco — APPROVED** [review]
✨🏆✨

**Comment by Elijbet** [comment]
The PR doesn't address the `underline` token gradient. Looks like `themed` needs to be updated to handle that case. 
```
  background-image: linear-gradient(currentColor, currentColor),
    linear-gradient(
      var(--calcite-link-underline-color-start, var(--calcite-color-brand-underline)),
      var(--calcite-link-underline-color-end, var(--calcite-color-brand-underline))
    );
    ```

**Pull request merged**